src/test/scala/chiselTests/ConstSpec.scala

```diff
@@ -0,0 +1,40 @@
+package chiselTests
+
+import chisel3._
+import chiselTests.{ChiselFlatSpec, Utils}
+import chisel3.experimental.BundleLiterals.AddBundleLiteralConstructor
+import chisel3.experimental.VecLiterals.AddVecLiteralConstructor
+import circt.stage.ChiselStage
+
+class ConstSpec extends ChiselFlatSpec with Utils {
+
+  "Const modifier on a wire or register" should "emit FIRRTL const descriptors" in {
+    val chirrtl = ChiselStage.emitCHIRRTL(new Module {
+      val foo = Wire(Const(UInt(8.W)))
+      val bar = Reg(Const(SInt(4.W)))
```

Comment:
Constant register should be disallowed. Correct me if I'm wrong, @dtzSiFive.

Reply:
👍 @trilorez confirms const reg is illegal. I will enforce from Chisel side.